openwisp_utils/admin_theme/dashboard.py

```diff
@@ -198,15 +203,18 @@ def get_dashboard_context(request):
             value['filters'] = filters
 
     # dashboard templates
+    extra_config = {}
     templates = []
     css = []
     js = []
-    for position, template_config in DASHBOARD_TEMPLATES.items():
-        templates.append(template_config['template'])
-        if 'css' in template_config:
-            css += list(template_config['css'])
-        if 'js' in template_config:
-            js += list(template_config['js'])
+    for _, template_config in DASHBOARD_TEMPLATES.items():
+        templates.append(template_config[0]['template'])
+        if 'css' in template_config[0]:
+            css += list(template_config[0]['css'])
+        if 'js' in template_config[0]:
+            js += list(template_config[0]['js'])
+        if template_config[1]:
+            extra_config.update(template_config[1])
```

Comment:
I'd just replace it to avoid ambiguity, `extra_config = template_config[1]`, the use of `update()` makes it look as if other parts of the code modify this.

Reply:
There is a slight problem in this change.
Consider:
In module 1 we have:
```
        register_dashboard_template(
            position=0,
            config={
                'template': 'dashboard_test.html',
                'css': ('dashboard-test.css',),
                'js': ('dashboard-test.js',),
            },
            extra_config={'test_extra_config': 'dashboard-test.config'},
        )
```
in module 2, we have:
```
        register_dashboard_template(
            position=0,
            config={
                'template': 'dashboard_test.html'
            },
            extra_config={'test_extra_config2': 'dashboard-test2.config'},
        )
 ```

Then the value of `test_extra_config` would be lost with `extra_config = template_config[1]`.
I don't think that's desirable. 
`extra_config.update(template_config[1])` fixes that problem! 😄

Reply:
why would you want to have 2 modules overriding and extending the exra conf? What problem does that solve?

Reply:
> why would you want to have 2 modules overriding the same template and extending the exra conf? What problem does that solve?

I don't think it's already been done anywhere, but I sure can think that there maybe a template in future which might want behave differently given different arguments and is used multiple times in a page.
In anycase, since it's not required for now, I am changing it to the recommended change.